src/target/ext/catch_syscalls.rs

```diff
@@ -0,0 +1,50 @@
+//! Enable or disable catching syscalls from the inferior process.
+
+use crate::arch::Arch;
+use crate::target::{Target, TargetResult};
+
+/// Target Extension - Enable and disable catching syscalls from the inferior
+/// process.
+///
```

Comment:
Could you also include a brief note mentioning the associated GDB client command (i.e: `catch syscalls`) that this command corresponds to? Linking back to https://sourceware.org/gdb/onlinedocs/gdb/Set-Catchpoints.html would also be a good idea.